# Qiime2_16S_analysis_basic
This repository serves as a Skelton of our basic 16S pipeline

# **QIIME2 Updated 4/21/20*
This pipeline is based on QIIME2 2019.10
And follows the Moving Pictures tutorial that can be found here. [Moving Picutres Tutorial](https://docs.qiime2.org/2019.10/tutorials/moving-pictures/)
QIIME2 has very good documentation and almost all questions can be answered here. [QIIME2](https://qiime2.org/)

#### **The scripts I have added here are to run QIIME2 analysis in the UCR cluster slurm framework.**
This allow the user to run scripts in the background without having to wait for resources.

## *Step1: Importing Data*
To run this batch script you will need to create a directory with all of you sequence files in a fastq.gz. 

Unless you edit the script the directory must be called **seqs**

To use this script file names must be in the Casava format ex. SampleID_L001_R1_001.fastq.gz

For importing other formats of data see. [QIIME2 importing data](https://docs.qiime2.org/2019.10/tutorials/importing/)


```
Qiime2_import_dada.sh
```
Input: Directory of sequencing files called seqs

Output1: demux-paired-end.qza

Output2: demux.qzv

**If you are having trouble and want to run this script outside of the queueing system see section 2**


## **Step2: Sequence quality control, feature table construction, and tree building**
This script combines multiple steps of the qiime pipeline.
In this step we use the program dada2 but there are other options. See QIIME2 documentation for other options and troubleshooting.
It will take a while so I suggest you use the batch script.

```
Qiime2_dada_and_tree.sh
```

## **Step3: Assign taxonomy**
This script assigns taxonomy using the Silva release 132 99% OTU database.
It will take a while so I suggest you use the batch script.


```
Qiime2_assign_taxonomy.sh
```

The rest of the qiime2 steps take far less time and resources and can be ran in an interactive environment.
## **First you need to enter and interactive node**
Do not run anything on a head node.

```
srun --x11 --mem=24gb --cpus-per-task 10 --ntasks 1 --time 24:00:00 --pty bash -l
```

## **Getting into a QIIME environment:**  
This is just a little trick to make sure that our cluster and QIIME2 are speaking the same language. 
Then entering the QIIME2 virtual environment

```
export LC_ALL=en_US.utf-8
export LANG=en_US.utf-8
conda init bash
source activate qiime2-2019.10
```
## **Generating diversity metrics:**
You may need to edit the sampling depth number based on your study.
```
qiime diversity core-metrics-phylogenetic --i-phylogeny rooted-tree.qza --i-table table-dada2.qza --p-sampling-depth 10000 --m-metadata-file sample-metadata.txt --output-dir core-metrics-results
```
## **Making taxa bar plots:**

```
qiime taxa barplot --i-table ./core-metrics-results/rarified-table.qza --i-taxonomy taxonomy.qza --m-metadata-file sample-metadata.txt --o-visualization taxa-bar-plots.qzv
````

